server/jamovi/tests/test_dataset.py

```diff
@@ -1,60 +1,214 @@
 """Tests for the dataset class."""
 
+from typing import Sequence
 from typing import TypeAlias
+from typing import Any
 import math
+import itertools
 
 import pytest
 
 from jamovi.server.dataset import DataSet
+from jamovi.server.dataset.duckdataset import DuckDataSet
 from jamovi.server.dataset import DataType
+from jamovi.server.dataset import MeasureType
+
+from .utils import equals
+from .utils import levels_must_equal
+from .utils import add_column_to_dataset
+from .utils import alter_levels
 
 
 NAN = float("nan")
 NAN_INT = -2147483648
 
-CellValue: TypeAlias = str | int | float | None
-
-
-def equals(x: CellValue, y: CellValue) -> bool:
-    """test if two cell values are equal"""
-    if isinstance(x, float) and isinstance(y, float):
-        if math.isnan(x):
-            return math.isnan(y)
-        if math.isnan(y):
-            return False
-        return pytest.approx(x) == y
-    else:
-        return x == y
+CellValue: TypeAlias = str | int | float
 
 
 def test_row_modification(simple_dataset: DataSet):
     """test modification of row count"""
-    simple_dataset.set_row_count(1537)
-    assert simple_dataset.row_count == 1537
+    simple_dataset.set_row_count(33)
+    assert simple_dataset.row_count == 33
+
+
+@pytest.mark.parametrize(
+    ("column_name",),
+    (("len",), ("dose",), ("supp",)),
+)
+def test_clear(toothgrowth_dataset: DuckDataSet, column_name: str):
+    """test column clear"""
+    ds = toothgrowth_dataset
+    column = ds[column_name]
+    column.clear()
+    for value in column:
+        if column.data_type is DataType.DECIMAL:
+            assert isinstance(value, float)
+            assert math.isnan(value)
+        elif column.data_type is DataType.TEXT:
+            assert value == ""
+        else:
+            assert value == NAN_INT
+
+    levels_must_equal(column.dlevels, [])
```

Comment:
Think it's best to do the actual assertion in the test itself and let the helper function just give you a boolean (like you do in the other helper functions)